pkg/trustedresources/verify.go

```diff
@@ -95,47 +93,23 @@ func VerifyResource(ctx context.Context, resource metav1.Object, k8s kubernetes.
 		}
 		return VerificationResult{VerificationResultType: VerificationError, Err: fmt.Errorf("failed to get matched policies: %w", err)}
 	}
-	objectMeta, signature, err := prepareObjectMeta(resource)
+	signature, err := extractSignature(resource)
 	if err != nil {
 		return VerificationResult{VerificationResultType: VerificationError, Err: err}
 	}
 	switch v := resource.(type) {
 	case *v1beta1.Task:
-		task := v1beta1.Task{
-			TypeMeta: metav1.TypeMeta{
-				APIVersion: "tekton.dev/v1beta1",
-				Kind:       "Task"},
-			ObjectMeta: objectMeta,
-			Spec:       v.TaskSpec(),
-		}
-		return verifyResource(ctx, &task, k8s, signature, matchedPolicies)
+		checksum, _ := v.CheckSum()
+		return verifyResource(ctx, v, k8s, checksum, signature, matchedPolicies)
```

Comment:
We already pass in the resource itself to `verifyResource`. Can we load the checksum later when needed to reduce number of args to the function?

Reply:
fixed.